hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -503,8 +493,8 @@ private String generateUniqueInstantTime(String initializationTime) {
     }
   }
 
-  private Pair<Integer, HoodieData<HoodieRecord>> initializePartitionStatsIndex(List<DirectoryInfo> partitionInfoList) {
-    HoodieData<HoodieRecord> records = HoodieTableMetadataUtil.convertFilesToPartitionStatsRecords(engineContext, partitionInfoList, dataWriteConfig.getMetadataConfig(), dataMetaClient,
+  private Pair<Integer, HoodieData<HoodieRecord>> initializePartitionStatsIndex(List<DirectoryInfo> partitionInfoList) throws IOException {
```

Comment:
I am ok to initialize new FSV while we are trying to initialize a new MDT partition. 
but lets not initialize new MDT FSV for regular updates. 
for eg, getRecordIndexReplacedRecords() should not be initializing a new MDT fsv and just re-use existing one. 

wdyt

Reply:
Hmm... any reason why was it being initialized in that method before? https://github.com/apache/hudi/blob/7530e4fa48fb6c32e9cafb587914521bbbb4bc23/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java#L1671

Reply:
not sure. probably we missed to notice that there is an existing MDT FSV that we can use.